src/test/java/il/ac/bgu/cs/bp/bpjs/analysis/SnapshotBenchmarks.java

```diff
@@ -0,0 +1,361 @@
+            //Iteration 1,starts already at request state A
+            for (int i = 0; i < NUM_STEPS; i++) {
+                next = sut.getUnvisitedNextNode(next, execSvc);
+                snapshots.add(next);
+            }
+            BProgramSyncSnapshotIO io = new BProgramSyncSnapshotIO(program);
+            execSvc.shutdown();
+
+            return snapshots.stream().map(node -> {
+                try {
+                    return io.serialize(node.getSystemState());
+                } catch (IOException e) {
+                    return new byte[0];
+                }
+            }).mapToInt(serializedSnap -> serializedSnap.length).toArray();
+
+        }
+
+        static long getVerificationTime(DfsBProgramVerifier vfr, BProgram prog) {
+            try {
+                VerificationResult res = vfr.verify(prog);
+                return res.getTimeMillies();
+            } catch (Exception e) {
+                return 0;
+            }
+        }
+
+        static long[] getVerificationTime(DfsBProgramVerifier vfr, String programPath, Map<String, Object> valueMap, int iteration_count) {
+            return LongStream.range(0, iteration_count).map(i -> {
+                BProgram prog = makeBProgram(programPath, valueMap);
+                return getVerificationTime(vfr, prog);
+            }).toArray();
+        }
+    }
+
+    /*
+        This benchmarks a program of variableSizedArrays
+        For verification time benchmarks, each test is run ITERATIONS times, with a larger array step every time.
+            These should be pretty much identical
+        For snapshot size benchmarks, each test is run once, and we measure the snapshot size for each step.
+            Snapshot size should grow linearly.
+     */
+    static class VariableSizedArraysBenchmark extends Benchmark {
+        static String IMPLEMENTATION = "benchmarks/variableSizedArrays.js";
+        static String TEST_NAME = "variableSizedArrays";
+        static int INITIAL_ARRAY_SIZE = 1;
+        static int ARRAY_STEP = 5; //Size of each step, how many objects are we adding
+        static int NUM_STEPS = 10; // How many steps to take
+
+
+        static void benchmarkVariableSizedArrays() throws Exception {
+            BenchmarkResult integerResults = measureIntegerSizes();
+
+            BenchmarkResult objectResults = measureObjectSize();
+            outputBenchResults(integerResults);
+            outputBenchResults(objectResults);
+        }
+
+        private static BenchmarkResult measureIntegerSizes() throws Exception {
+            System.out.println("Measuring effect of integer size");
+            return measureProgram(0);
+        }
+
+        private static BenchmarkResult measureObjectSize() throws Exception {
+            System.out.println("Measuring effect of object size");
+            return measureProgram(1);
+        }
+
+        private static BenchmarkResult measureProgram(int object_type) throws Exception {
```

Comment:
`object_type` being an `int` is a bit of a "magic number" - can we use Java's enums here? (also, why snake_case instead of camelCase in this parameter?)

Reply:
(Will fix)
Not sure how Java enums translate to the JS vars, need to test that